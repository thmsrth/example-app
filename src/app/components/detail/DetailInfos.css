.DetailInfos {
  box-shadow: var(--section-box-shadow);
  margin: var(--section-margin);
  background-color: var(--section-background-color);
  border-top: var(--section-border-top);
  border-radius: var(--section-border-radius);
  width: calc(33.33% - (var(--section-margin) * 2));
  flex-basis: calc(33.33% - (var(--section-margin) * 2));
}

.DetailInfos__container {
  padding: var(--section-padding) 0 0;
}

.DetailInfos__header {
  padding: var(--section-header-padding);
  display: flex;
  justify-content: space-between;
  border-bottom: var(--section-header-border-bottom);
}

.DetailInfos__title {
  font-size: var(--section-title-font-size);
  margin: var(--section-title-margin);
}

.DetailInfos__list {
  padding: var(--section-padding);
}

.DetailInfos__propertyValue {
  word-break: break-all;
}
