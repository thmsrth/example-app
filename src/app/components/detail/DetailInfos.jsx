import React from 'react';

import dataItemType from '../types/dataItem';

import DetailsList from '../../../global/components/DetailsList/DetailsList';

import I18n from '../../utils/I18n';

import './DetailInfos.css';

const DetailInfos = ({ item }) => {
  const keys = [
    {
      name: 'id',
      title: `${I18n.t('en.data.attribute.id.title')}:`,
      type: 'number',
    },
    {
      name: 'name',
      title: `${I18n.t('en.data.attribute.name.title')}:`,
      type: 'number',
    },
    {
      name: 'fat',
      title: I18n.t('en.data.attribute.fat.title'),
      type: 'number',
    },
    {
      name: 'protein',
      title: I18n.t('en.data.attribute.protein.title'),
      type: 'number',
    },
    {
      name: 'calories',
      title: I18n.t('en.data.attribute.calories.title'),
      type: 'number',
    },
    {
      name: 'status',
      title: I18n.t('en.data.attribute.status.title'),
      type: 'text',
    },
    {
      name: 'createdAt',
      title: I18n.t('en.data.attribute.createdAt.title'),
      type: 'datetime',
      format: 'DD/MM/YYYY HH:mm',
    },
    {
      name: 'createdBy',
      title: I18n.t('en.data.attribute.createdBy.title'),
      type: 'text',
    },
    {
      name: 'updatedAt',
      title: I18n.t('en.data.attribute.updatedAt.title'),
      type: 'datetime',
      format: 'DD/MM/YYYY HH:mm',
    },
    {
      name: 'updatedBy',
      title: I18n.t('en.data.attribute.updatedBy.title'),
      type: 'text',
    },
  ];

  if (item && item.id) {
    return (
      <div className="DetailInfos">
        <div className="DetailInfos__container">
          <div className="DetailInfos__header">
            <h3 className="DetailInfos__title">
              {I18n.t('en.data.infos.title')}
            </h3>
          </div>
          <DetailsList
            item={item}
            keys={keys}
            className="DetailInfos"
          />
        </div>
      </div>
    );
  }
  return null;
};

export default DetailInfos;

DetailInfos.propTypes = {
  item: dataItemType,
};

DetailInfos.defaultProps = {
  item: null,
};
