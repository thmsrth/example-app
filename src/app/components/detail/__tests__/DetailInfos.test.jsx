import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import staticData from '../../../static/data.json';
import DetailInfos from '../DetailInfos';

global.permissions = {};

const item = staticData._embedded.foods[0];
const props = {
  item,
};

describe('DetailInfos', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<DetailInfos {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
