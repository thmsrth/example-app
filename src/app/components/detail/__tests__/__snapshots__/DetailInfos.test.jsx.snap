// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailInfos should render the component correctly 1`] = `
<div
  className="DetailInfos"
>
  <div
    className="DetailInfos__container"
  >
    <div
      className="DetailInfos__header"
    >
      <h3
        className="DetailInfos__title"
      >
        Infos
      </h3>
    </div>
    <DetailsList
      className="DetailInfos"
      item={
        Object {
          "calories": 266,
          "createdAt": "2018-10-28T14:48:54.01293",
          "createdBy": "thomas",
          "fat": 34,
          "id": 1,
          "name": "Pizza",
          "protein": 0.6,
          "status": "UNHEALTHY",
          "updatedAt": null,
          "updatedBy": null,
        }
      }
      keys={
        Array [
          Object {
            "name": "id",
            "title": "ID:",
            "type": "number",
          },
          Object {
            "name": "name",
            "title": "Name:",
            "type": "number",
          },
          Object {
            "name": "fat",
            "title": "Fat",
            "type": "number",
          },
          Object {
            "name": "protein",
            "title": "Protein",
            "type": "number",
          },
          Object {
            "name": "calories",
            "title": "Calories",
            "type": "number",
          },
          Object {
            "name": "status",
            "title": "Status",
            "type": "text",
          },
          Object {
            "format": "DD/MM/YYYY HH:mm",
            "name": "createdAt",
            "title": "Created at",
            "type": "datetime",
          },
          Object {
            "name": "createdBy",
            "title": "Created by",
            "type": "text",
          },
          Object {
            "format": "DD/MM/YYYY HH:mm",
            "name": "updatedAt",
            "title": "Updated at",
            "type": "datetime",
          },
          Object {
            "name": "updatedBy",
            "title": "Updated by",
            "type": "text",
          },
        ]
      }
      propertyFromData={null}
    />
  </div>
</div>
`;
